lib/ruby_indexer/test/index_test.rb

```diff
@@ -1672,6 +1672,26 @@ def test_linearizing_singleton_object
       )
     end
 
+    def test_extend_self
+      @index.index_single(IndexablePath.new(nil, "/fake/path/foo.rb"), <<~RUBY)
+        module Foo
+          def bar
+          end
+
+          extend self
+
+          def baz
+          end
+        end
+      RUBY
+
+      ["bar", "baz"].product(["Foo", "Foo::<Class:Foo>"]).each do |method, receiver|
+        entry = @index.resolve_method(method, receiver)&.first
+        refute_nil(entry)
+        assert_equal(method, T.must(entry).name)
+      end
```

Comment:
Let's also add an assertion for the expected linearized ancestors, which should include `Foo` itself after the singleton class. This will make it easier to understand what the expected inheritance structure is.

```suggestion
      end

      assert_equal(
        [
          "Foo::<Class:Foo>",
          "Foo",
          "Module",
          "Object",
          "Kernel",
          "BasicObject",
        ],
        @index.linearized_ancestors_of("Foo::<Class:Foo>"),
      )
```

Reply:
Will do!